### ISSUE TITLE
Selected state of the Nav bar is missing

### ISSUE THREAD

**Issue by schne324** [dequelabs/cauldron#593]
<dl><dt>Description</dt><dd>The selected state of an element is not conveyed to screen reader users.
Selected state of the Nav bar item is missing. 
Element - NavItem 3 </dd><dt>Impact</dt><dd>Serious</dd><dt>Recommendation</dt><dd>RULE : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

HOW TO FIX: 
Fix this issue by using the aria-current attribute to indicate which element is the current item within a container or set of related elements, such as the currently selected page, step, location, date, or time.

REFERENCE: 
WAI-ARIA Recommendation: https://www.w3.org/TR/wai-aria-1.1/#aria-current

BACKGROUND: 
States and properties are attributes used to convey essential information about an element to screen readers and other assistive technologies. Some roles require certain state and property information - such as the checked/unchecked state of a checkbox. Native HTML elements provide those required states and properties, so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant states and properties using ARIA.</dd><dt>Source code</dt><dd>

```html
<li class="NavItem NavItem--active"><a href="#">NavItem 3</a></li>
```
</dd><dt>Checkpoint</dt><dd>Name, Role, Value (4.1.2.a)</dd><dt>URL</dt><dd>https://cauldron.dequelabs.com/components/NavBar</dd></dl>

**Comment by schne324** [comment]
set `aria-current="page"` on active pages (based on [`active` prop of `<NavItem />`](https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/components/NavBar/NavItem.tsx#L7))

**Comment by scurker** [comment]
This should be fixed now
